File: 3351-maximize-happiness-of-selected-children/solution.cs
using System;

public class Solution {
    public long MaximumHappinessSum(int[] happiness, int k) {
        Array.Sort(happiness);
        
        long total = 0;
        int n = happiness.Length;
        
        for (int i = 0; i < k; i++) {
            int currentVal = happiness[n - 1 - i] - i;
            if (currentVal <= 0) break;
            
            total += (long)currentVal;
        }
        
        return total;
    }
}

File: 3351-maximize-happiness-of-selected-children/solution.kt
class Solution {
    fun maximumHappinessSum(happiness: IntArray, k: Int): Long {
        happiness.sort()
        
        var total: Long = 0
        val n = happiness.size
        
        for (i in 0 until k) {
            val currentVal = happiness[n - 1 - i] - i
            
            if (currentVal <= 0) break
            
            total += currentVal.toLong()
        }
        
        return total
    }
}
